File: ReviewService/src/main/java/com/clarivate/reviewservice/Service/ReviewerService.java
package com.clarivate.reviewservice.Service;
import com.clarivate.reviewservice.dto.Request.ReviewCommentRequest;
import com.clarivate.reviewservice.dto.Request.ReviewRecommendationRequest;
import com.clarivate.reviewservice.dto.Response.ReviewCommentResponse;
import com.clarivate.reviewservice.dto.Response.ReviewProcessResponse;

import java.util.List;

public interface ReviewerService {
    List<ReviewProcessResponse> getAssignedReviews(Long reviewerId);
    ReviewCommentResponse addComment(ReviewCommentRequest request);
    ReviewProcessResponse submitRecommendations(Long reviewId, ReviewRecommendationRequest request);

    List<ReviewCommentResponse> getComments(Long reviewId);
}

File: research-frontend-main/src/app/core/services/reviewer.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '@env/environment';
import {
  ReviewProcessResponse,
  ReviewCommentRequest,
  ReviewCommentResponse,
  ReviewRecommendationRequest
} from '@core/models';

@Injectable({ providedIn: 'root' })
export class ReviewerService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiUrls.review}/reviewer`;

  getAssignedReviews(reviewerId: number): Observable<ReviewProcessResponse[]> {
    return this.http.get<ReviewProcessResponse[]>(`${this.baseUrl}/${reviewerId}/reviews`);
  }

  addComment(request: ReviewCommentRequest): Observable<ReviewCommentResponse> {
    return this.http.post<ReviewCommentResponse>(`${this.baseUrl}/comments`, request);
  }

  submitRecommendation(reviewId: number, request: ReviewRecommendationRequest): Observable<ReviewProcessResponse> {
    return this.http.put<ReviewProcessResponse>(`${this.baseUrl}/reviews/${reviewId}/recommendation`, request);
  }

  getComments(reviewId: number): Observable<ReviewCommentResponse[]> {
    return this.http.get<ReviewCommentResponse[]>(`${this.baseUrl}/reviews/${reviewId}/comments`);
  }
}
